collectors/ciscoamp/utils.js

```diff
@@ -32,11 +33,7 @@ function getAPILogs(baseUrl, authorization, apiUrl, state, accumulator, maxPages
                     }
                     else {
                         if (state.stream === Events && accumulator.length > 0) {
-                            // Api return the data in desending order, so set new start date form first record if not available pull date from fallup records.
-                            newSince = accumulator[0].date ? accumulator[0].date : accumulator[1].date ? accumulator[1].date : accumulator[2].date;
```

Comment:
I think we need to keep this.

Reply:
if there is no date available we can take from Until value from state object

Reply:
this code was added to make sure we find `date` property in most recently collected messages. This was done to make sure `newSince` moves, otherwise it'll result in recollections. 
I think it should be
```
// Api return the data in desending order, so set new start date form first record if not available pull date from fallup records.
  findNewSince = accumulator[0].date ? accumulator[0].date : accumulator[1].date ? accumulator[1].date : accumulator[2].date;
  if (!findNewSince) {
      reject(`CAMP000005 Date is not available in Events api response`);
  }
  newSince = moment(findNewSince).add(1, 'seconds').toISOString();
```